src/rules/at-import-partial-extension/index.js

```diff
@@ -82,15 +84,20 @@ function rule(expectation, _, context) {
         if (extension && isScssPartial && expectation === "never") {
           if (context.fix) {
             const extPattern = new RegExp(`\\.${extension}(['" ]*)$`, "g");
-            decl.params = decl.params.replace(extPattern, "$1");
+            atRule.params = atRule.params.replace(extPattern, "$1");
 
             return;
           }
 
+          const dotExt = `.${extension}`;
+          const index =
+            atRuleParamIndex(atRule) + atRule.params.lastIndexOf(dotExt);
+
           utils.report({
             message: messages.rejected(extension),
-            node: decl,
-            word: extension,
+            node: atRule,
+            index,
+            endIndex: index + dotExt.length,
```

Comment:
[note] I think the end position should include a dot (`.`) since the message also includes it:

```
Unexpected extension ".scss" in @import
```